lib/sycamore/sycamore/transforms/aggregation.py

```diff
@@ -0,0 +1,259 @@
+from typing import Callable, Optional, TYPE_CHECKING, Union
+
+from sycamore.data.document import split_data_metadata
+from sycamore.plan_nodes import UnaryNode, Node
+from sycamore.data import Document, MetadataDocument
+
+from sycamore.utils.lineage_utils import update_lineage
+from sycamore.utils.thread_local import ADD_METADATA_TO_OUTPUT, ThreadLocal
+
+if TYPE_CHECKING:
+    from ray.data import Dataset
+    import numpy as np
+
+
+class AggregationNode(UnaryNode):
+    def __init__(
+        self,
+        child: Optional[Node],
```

Comment:
You have an Aggregation class which defines name ... zero_factory. I'd just take such an object as an argument.